runners/katacoda/index.ts

```diff
@@ -228,6 +228,26 @@ export class Katacoda extends Runner {
 
     }
 
+    runCloneRepository(step: Step, command: Command): RunResult {
+
+        let cdCommand = this.changeCurrentDir(path.join("/root", "devonfw", "workspaces", "main"));
+        let directoryPath = path.join(command.parameters[0]).replace(/\\/g, "/");
+
+        this.steps.push({
+            "title": "Clones Repository " + command.parameters[1],
+            "text": "step" + this.stepsCount + ".md"
+        });
+
+        // checks if path is an empty string
+        if(Boolean(command.parameters[0].trim())) {
+            this.renderTemplate("cloneRepository.md", this.outputPathTutorial + "step" + (this.stepsCount++) + ".md", { text: step.text, textAfter: step.textAfter, cdCommand: cdCommand, directoryPath: directoryPath, repository: command.parameters[1]});
+        } else {
+            this.renderTemplate("cloneRepositoryEmptyString.md", this.outputPathTutorial + "step" + (this.stepsCount++) + ".md", {text: step.text, textAfter: step.textAfter, cdCommand: cdCommand, repository: command.parameters[1]});
+        }
```

Comment:
The second template is not needed. You can use if statements in the ejs templates. So you can check in the template itself if the given string is not empty and render then some additional text.